evalml/automl/engine/cf_engine.py

```diff
@@ -0,0 +1,140 @@
+from evalml.automl.engine.engine_base import (
+    EngineBase,
+    EngineComputation,
+    evaluate_pipeline,
+    score_pipeline,
+    train_pipeline,
+)
+
+
+class CFClient:
+    """Custom CFClient API to match Dask's CFClient and allow context management."""
+
+    def __init__(self, pool):
+        self.pool = pool
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, typ, value, traceback):
```

Comment:
Typically the exit method of a context manager does some clean-up: Should we call self.pool.close() here?